analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Binary.cs

```diff
@@ -552,5 +552,64 @@ public void BinaryNotEqualsNull_SetsBoolConstraint_Unknown_ComparedToNull_VB(str
                 .And.ContainSingle(state => state.SymbolsWith(BoolConstraint.True).Any(x => x.Name == "Value") && state.SymbolsWith(ObjectConstraint.NotNull).Any(x => x.Name == "Arg"))
                 .And.ContainSingle(state => state.SymbolsWith(BoolConstraint.False).Any(x => x.Name == "Value") && state.SymbolsWith(ObjectConstraint.Null).Any(x => x.Name == "Arg"));
         }
+
+        private static IEnumerable<object[]> NullableRelationalNullExpressions => new object[][]
+        {
+            new string[] { "arg >= null" },
+            new string[] { "arg > null" },
+            new string[] { "arg < null" },
+            new string[] { "arg <= null" },
+            new string[] { "null >= arg" },
+            new string[] { "null > arg" },
+            new string[] { "null < arg" },
+            new string[] { "null <= arg" },
+            new string[] { "arg > (int?)null" },
+            new string[] { "arg > new Nullable<int>()" },
+            new string[] { "arg > (null as int?)" },
+            new string[] { "arg > nullValue" },
+            new string[] { "nullValue > arg" },
+            new string[] { "nullValue > 42" },
+            new string[] { "nullValue > notNullValue" },
+            new string[] { "notNullValue > nullValue"},
+        };
+
+        [DataTestMethod]
+        [DynamicData(nameof(NullableRelationalNullExpressions))]
+        public void Binary_NullableRelationalNull_IfUnreachable_CS(string expression)
+        {
+            var code = $$"""
+                int? notNullValue = 42;
+                int? nullValue = null;
+                if ({{expression}})
+                {
+                    Tag("If_Unreachable");
+                }
+                else
+                {
+                    Tag("Else");
+                }
+                Tag("End");
+                """;
+
+            var validator = SETestContext.CreateCS(code, ", int? arg").Validator;
+            validator.ValidateContainsOperation(OperationKind.Binary);
+            validator.ValidateTagOrder("Else", "End");
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(NullableRelationalNullExpressions))]
+        public void Binary_NullableRelationalNull_SetsBoolConstraint_CS(string expression)
+        {
+            var code = $$"""
+                int? notNullValue = 42;
+                int? nullValue = null;
+                var value = {{expression}};
+                Tag("Value", value);
+                """;
+
+            var validator = SETestContext.CreateCS(code, ", int? arg").Validator;
+            validator.ValidateContainsOperation(OperationKind.Binary);
+            validator.ValidateTag("Value", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.False));
+        }
```

Comment:
This doesn't need to be 2nd test. Just add `value` and `Tag("Value", value)` to the previous one. And then you can remove the `DynamicData` thing that is hard to maintain (needs to be at the beginning of the file to follow coding style).

```suggestion
```